---
title: "Real-time estimation of the time-varying transmission advantage of Omicron in England using S-Gene Target Status as a Proxy"
subtitle: "Summary report"
author: Sam Abbott (1), Katharine Sherratt (1), Sebastian Funk (1)
bibliography: library.bib
csl: https://raw.githubusercontent.com/citation-style-language/styles/master/apa-numeric-superscript-brackets.csl
date: "`r format(Sys.Date(), format = '%B %d, %Y')`"
output:
  html_document:
    theme: cosmo
    toc: true
    toc_float: true
    toc_depth: 4
    includes:
      before_body: header.html
      after_body:  footer.html
  pdf_document:
    extra_dependencies: ["float"]
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, include = TRUE,
                      warning = FALSE, message = FALSE,
                      root.dir = here::here(),
                      fig.width = 12, fig.height = 9)
options(digits = 1)
library(here)
library(forecast.vocs)
library(dplyr)
library(loo)
library(scoringutils)
library(knitr)
library(data.table)
```

```{r load-results}
# load packages
library(ggplot2)
library(scales)
library(dplyr)
library(here)
library(readr)
library(data.table)

# load functions
source(here("R", "load-local-data.R"))
source(here("R", "plot-summary.R"))
source(here("R", "plot-daily-cases.R"))
source(here("R", "plot-omicron-95.R"))
source(here("R", "plot-cumulative-percent.R"))

# get latest date
date_latest <- get_latest_date()

# load data
daily <- load_local_data(date_latest)
pop <- load_population()

# load results
sgtf <- load_results(date_latest)
bias <- load_results(date_latest, type = "bias")

sgtf$posterior <- sgtf$posterior[,
   variant_relationship := stringr::str_to_sentence(variant_relationship)
]
sgtf_posterior <- sgtf$posterior[
  variant_relationship == "Correlated"
]

bias_posterior <- bias$posterior[variant_relationship == "correlated"]
```

```{r summarise-results}
# get cases from correlated model
cases <- summary(sgtf_posterior, type = "cases")

# get VOC fraction from correlated model
voc_frac <- summary(sgtf_posterior, type = "voc_frac")

# cumulative percentage with each variant
cum_per <- cumulative_percentage(cases, pop) %>%
  filter(type %in% "Omicron")
```

```{r setup-dates}
# set up dates
date_data_start <- min(as.data.table(sgtf$data)[!is.na(seq_total)]$date)
date_forecast_start <- max(cases$date) - 2 * 7
date_forecast_end <- max(cases$date)
```

1. Centre for the Mathematical Modelling of Infectious Diseases, London School of Hygiene & Tropical Medicine, London WC1E 7HT, United Kingdom

**Estimates are updated daily as new data is released but the summary of the situation is updated less frequently and so may not match current estimates. All data (both raw data and estimates) are available [here](https://github.com/epiforecasts/omicron-sgtf-forecast/tree/main/data) and the report should be fully reproducible. Reports and data as available at the time of release are available from the [release page](https://github.com/epiforecasts/omicron-sgtf-forecast/releases). See our [news file](https://github.com/epiforecasts/omicron-sgtf-forecast/blob/main/NEWS.md) for details of what updates were made when.**

## Introduction

Since being highlighted by scientists in South Africa Omicron has spread rapidly globally. Using initial South African data, it has been estimated that Omicron may be both more transmissible and have greater immune escape than the previously dominant Delta variant [@pearson-omicron]. 

In this work, we use S-gene target failure (SGTF) as a proxy of variant status combined with reported case counts to explore the evidence for changes in transmission advantage over time for the Omicron variant. If present this could indicate the impact of immune escape, sampling bias in SGTF data or differences in the populations within which the variants are circulating. We also report estimates for growth rates by variant and overall, case counts overall and by variant for a 14 day forecast window assuming constant future growth, the date at which Omicron will become dominant in England and in each UKHSA region, and the estimated cumulative percentage of the population with a reported Omicron case. We also explore the potential for bias in S-gene target sampling by comparing cases that have a reported SGTF status to cases with an unknown SGTF status. 

## Methods

### Data

We use S-gene status by specimen date sourced from UKHSA as a direct proxy for the Omicron variant with a target failure indicating a case has the Omicron variant. We augment this data with reported cases counts by date of specimen truncated by two days to account for delayed reporting. Data was available for England and for UKHSA region from both sources.

### Models

We consider two autoregressive models with different assumptions about the relationship between growth rates for Omicron and non-Omicron cases. Both models estimate expected cases for each variant as a combination of expected cases from the previous day and the exponential of the log growth rate. The growth rate is then itself modelled as a differenced AR(1) process. 

For the first model variants are assumed to have growth rates related by a fixed scaling (described from now on as the scaled model). In this model, variant growth rates then share a single differenced AR(1) process meaning that they vary over time in the same way. This model assumes that variants differ only due to a transmission advantage and that there are no time-varying biases in the reported data.

In the second model, we relax the assumption that variants co-vary using a vector autoregression structure which assume that the 1st differences of the variant growth rates are drawn from a multivariate normal distribution. This model formulation can account for variant differences other than a transmission advantage and can also better handle time-varying biases in data sources. Crucially, in the absence of evidence that variants do not co-vary, it reduces to the co-varying model.

For both models, we fit jointly to reported cases and SGTF data assuming a negative binomial and beta-binomial observation model respectively. Day of the week reporting periodicty for case counts is captured using a random effect fo r the day of the week. We initialise both models by fitting to a week of case only data where the Omicron variant is assumed to not be present (from the 17th of November to the 22nd).

A full description of the models described here can be found in the documentation for the [`forecast.vocs` R package](https://epiforecasts.io/forecast.vocs/) [@forecast.vocs].

### Statistical Inference

We first visualised our combined data sources (cases by specimen date and SGTF status by specimen date). We then fit both models separately to data for England and to each UKHSA region. Using these model fits we report posterior estimates from the best fitting model for the following summary statistics of epidemiological interest: growth rates by variant and overall, the time-varying transmission advantage for the Omicron variant, case counts overall and by variant for a 14 day forecast window assuming constant future growth, the date at which Omicron will become dominant in England and in each UKHSA region, and the estimated cumulative percentage of the population with a reported Omicron case. 

We explored the potential for bias in S-gene target sampling by comparing cases that have a reported SGTF status to cases with an unknown SGTF status. We fit the correlated model to case counts and S-gene tested status and reported the apparent transmission advantage for those with a S-gene status over time. Any variation in this metric from 100% may be interpreted as indicating biased sampling of those with known S-gene status, which may bias our main results.

### Implementation

All models were implemented using the [`forecast.vocs` R package](https://epiforecasts.io/forecast.vocs/) [@R; @forecast.vocs] and fit using `stan` [@stan] and `cmdstanr` [@cmdstanr]. Each model was fit using 2 chains with each chain having 1000 warmup steps and 2000 sampling steps. Convergence was assessed using the Rhat diagnostic [@stan]. Models were compared using approximate leave-one-out (LOO) cross-validation [@loo; @loo-paper] where negative values indicate an improved fit for the correlated model.


## Limitations


* SGTF may be an imperfect proxy for Omicron. We make no adjustment for the background rate of SGTF observed for Delta or other factors that might lead to SGTF, or a confirmed detected S-gene observed with Omicron.

* The public case data used here does not include known reinfections. If the proportion of infections that are reinfections varies over time and between variants then this could affect our estimates.

* We make a crude adjustment for the right censoring of reported case counts by specimen date by excluding the last two days of data. However, this may be insufficient to correct for the bias and so our real-time growth rates estimates may be biased downwards.

* Our analysis is based on reported case counts augmented with SGTF data. Growth rates estimated from reported case counts may be biased during periods when changes in testing practice occur or when available testing is at capacity.

* We reported the log growth rate approximation to the growth rate. This approximation becomes less valid for high absolute growth rates.

* Our models do not attempt to capture the underlying transmission process or delays in reporting. In particular, we do not consider the generation time of variants which may differ. Differences in the generation time would result in apparent differences in the growth rate (on the scale of the generation time) even in scenarios where growth rates were fixed. 

* Our forecasts assume a fixed growth rate across the forecast horizon and do not account for population susceptibility or other factors such as changes in contact patterns.

* Our model only estimates an overall transmission advantage. We do not attempt to distinguish between "intrinsic" transmissibility and immune escape, a combination of which is likely to be underlying any sustained differences in growth rate.

* Our estimates are at the national or UKHSA region scale. There may be additional variation within these areas that these estimates cannot capture.

* We used an approximate form of leave-one-out cross validation to compare the performance of the models we considered. This may be inappropriate in situations where performance on held out time-series data is the target of interest.

## Results 

### Summary

**Last updated: 2021-12-31**

* A model that assumed that variants growth rates varied over time relative to each other performed better in all regions than a model where the relative difference was constant over time. 

* Prior to the introduction of Omicron the growth rate of non-Omicron cases was positive in most regions. In the last few weeks prior to Christmas the growth rate for non-Omicron cases became negative suggesting that if nothing changes Omicron will be the only circulating variant at some point in the future.

* The growth rate for Omicron cases has been consistently positive in all regions since the 23rd of November though with slightly different absolute values and with different variation over time.

* In the week before Christmas both case data and the SGTF data suggested that the growth rate of Omicron in reported cases has reduced markedly though still appeared positive in all regions excluding London.

* Since Christmas growth rates for both variants have slightly increased though only Omicron has a positive growth rate. The estimated growth rate for Omicron in London returned to being positive on the 31st of December.

* Assuming a fixed difference in growth between variants indicates an advantage for Omicron in the order of 30% to 50% across all regions. Allowing this to vary (which was shown to better represent the data) implies that this advantage generally appears to have increased over time until the week before Christmas where it fell back to initially observed levels or below in all regions. 

* The variation in transmission advantage also differed by region with some regions having a fairly stable trend (such as the South West) and some (such as the North East) indicating large changes.

* Assuming growth rates stay as currently estimated we forecast that approximately 5% to 10% of the English population will have reported an Omicron case by the 10th of January 2021, although with significant regional variation. This does not take into account any changes in testing behaviour, capacity, or changes in growth rate due to susceptible depletion or other interventions.

* We found some evidence of bias in sampling S-gene status though this varied across regions and over time. There was particularly strong evidence of biased sampling over time in the East Midlands and in the more recent data.

### Latest available estimates using data up to: `r date_forecast_start`

```{r summary-plot, fig.width = 10, fig.height = 6, fig.cap = "Summary of latest estimates for England and by UKHSA region, showing: bias in S-gene sampling, comparing the Omicron transmission advantage among cases with any SGT result, against cases with no SGT result (at 1); cumulative % of population reported as a case with Omicron variant; current growth rate, and current transmission advantage of Omicron variant. 90% and 30% credible intervals are shown."}
plot_summary(sgtf_posterior = sgtf_posterior,
             cum_per = cum_per,
             bias_posterior = bias_posterior,
             date_forecast_start = date_forecast_start)
```

### Data description

```{r data, fig.cap = "Daily cases in England and by UKHSA region, with S-gene target result (failed, confirmed detected, or unknown), and centred 7-day moving average up to date of data truncation (dotted line). Source: UKHSA and coronavirus.gov.uk; data by specimen date."}
plot_daily_cases(
  daily, truncate_date = date_forecast_start, caption = "",
  start_date = date_data_start, smooth_total = TRUE
)
```

### Model comparison

```{r compare-score}
scores <- janitor::clean_names(
  select(sgtf$loo, -forecast_date), case = "sentence"
)
kable(scores, caption = "Estimated differences in the ELPD metric (with
 standard errors) between the scaled and correlated models for England and UKHSA
 regions. Negative values indicate that the correlated model is estimated to be
 a better fit to the data. ")
```

### Growth rate of reported cases overall, with the Omicron variant, and not with the Omicron variant

```{r growth, fig.cap = "Estimates of the time-varying daily growth rate for England and by UKHSA region (using the log scale approximation) for reported cases overall, cases with the Omicron variant, and non-Omicron cases. Growth rates are assumed to be piecewise constant with potential changes every 3 days. Growth rates are assumed to be constant beyond the forecast horizon (vertical dashed line). 90% and 60% credible intervals are shown."}
plot_growth(sgtf_posterior) +
  facet_wrap(~ region)
```

### Proportion of cases with the Omicron variant

#### Natural scale

```{r voc-frac-natural, fig.cap = "Estimates of the proportion of reported cases with the Omicron variant for England and by UKHSA region using both the scaled model and the correlated model on the natural scale. The forecast horizon is indicated using a dashed line. 90% and 60% credible intervals are shown. Observation error is not shown meaning that the posterior predictions may look overly precise."}
plot(
  sgtf$posterior, as.data.table(sgtf$data), type = "voc_frac",
  fill = variant_relationship, voc_label = "Omicron variant", log = FALSE
) +
  labs(fill = "Variant relationship") +
  facet_wrap(~ region)
```

#### Logit scale

```{r voc-frac-logit, fig.cap = "Estimates of the proportion of reported cases with the Omicron variant for England and by UKHSA region using both the scaled model and the correlated model on the logit scale. Note that scaled model is effectively assuming a linear relationship on this scale. The forecast horizon is indicated using a dashed line. 90% and 60% credible intervals are shown. Observation error is not shown meaning that the posterior predictions may look overly precise."}
plot(
  sgtf$posterior, as.data.table(sgtf$data), type = "voc_frac",
  fill = variant_relationship, voc_label = "Omicron variant"
) +
  labs(fill = "Variant relationship") +
  facet_wrap(~ region)
```

### Date at which Omicron estimated to account for 95% of reported cases
```{r omicron-95, fig.cap = "Estimated dates at which 95% of reported cases will have the Omicron variant for England and UKHSA regions. 90% and 60% credible intervals are shown along with the median estimate (point),", fig.height = 6, fig.width = 6}
plot_omicron_95(
  voc_frac = voc_frac,
  forecast_start = date_forecast_start,
  forecast_end = date_forecast_end
)
```


### Time-varying transmission advantage of the Omicron variant

```{r voc-advantage, fig.cap = "Estimates of the time-varying daily transmission advantage for the Omicron cases vs non-Omicron cases for England and by UKHSA region. Estimates are shown for both the scaled model and the correlated model. Variation in the advantage over time could be an indicator of biases in the SGTF data, differences in the populations in which the variants are circulating or as a result of immune escape. Transmission advantage is assumed to be constant beyond the forecasting horizon (dashed line). 90% and 60% credible intervals are shown. Advantage is calculated as the exponential of Omicron's growth rate minus non-Omicron's growth rate."}
plot(
  sgtf$posterior, type = "voc_advantage",
  fill = variant_relationship, group = variant_relationship,
  voc_label = "Omicron variant"
) +
  labs(fill = "Variant relationship",
       y = "Transmission advantage for the Omicron variant"
  ) +
  facet_wrap(~ region)
```

### Posterior predictions and forecasts of reported cases

#### Natural

```{r natural-cases, fig.cap = "Estimates of the number of reported cases by variant and overall for England and each UKHSA region. Estimates are shown both within the window supported by the data and for 14 days afterwards as a forecast where it is assumed that growth rates for each variant are stable. The forecast horizon is indicated using a dashed line. 90% and 60% credible intervals are shown."}
plot(
  sgtf_posterior, as.data.table(sgtf$data), type = "cases", log = FALSE
) +
  facet_wrap(~region, scales = "free_y")
```

#### Log

```{r log-cases, fig.cap = "Estimates of the number of reported cases by variant and overall for England and each UKHSA region on the log 2 scale. Estimates are shown both within the window supported by the data and for 14 days afterwards as a forecast where it is assumed that growth rates for each variant are stable. The forecast horizon is indicated using a dashed line. 90% and 60% credible intervals are shown.", fig.width = 12}
plot(
  sgtf_posterior, as.data.table(sgtf$data), type = "cases"
) +
  facet_wrap(~region, scales = "free_y")
```

### Cumulative percentage of the population with a reported Omicron case

```{r cumulative-population, fig.cap = "Estimates of the cumulative percentage of the population with reported Omicron cases in England and by UKHSA. The forecast horizon is indicated using a dashed line. 90% and 60% credible intervals are shown."}
plot_cumulative_percent(
  cum_per, forecast_start = date_forecast_start, data_start = date_data_start
)
```

### Evidence of sampling bias in S-gene target results among Covid-19 test-positive cases

```{r, bias-advantage, eval = TRUE, fig.cap = "Estimates of the time-varying daily transmission advantage for cases with a reported S-gene status vs those without for England and by NHS region. Deviation from 100% (dashed horizontal line) indicates that growth rates differ in the two populations which is an indication of sampling bias. If present this sampling bias could impact our main results. Transmission advantage is assumed to be constant beyond the forecasting horizon (dashed line vertical line). 90% and 60% credible intervals are shown. Note that here a 7-day piecewise constant growth rate has been used."}
plot(bias$posterior, type = "voc_advantage") +
  geom_hline(yintercept = 1, linetype = 3, alpha = 0.8) +
  labs(y = "Transmission advantage of cases with a reported S gene
            status versus those without") +
  facet_wrap(~ region)
```

## References


---
title: "Estimation of the test to test distribution as a proxy for generation interval distribution for the Omicron variant in England"
author: Sam Abbott (1), Katharine Sherratt (1), Moritz Gerstung (2), Sebastian Funk (1)
bibliography: library.bib
csl: https://raw.githubusercontent.com/citation-style-language/styles/master/apa-numeric-superscript-brackets.csl
date: "`r format(Sys.Date(), format = '%B %d, %Y')`"
output:
  html_document:
    theme: cosmo
    toc: true
    toc_float: true
    toc_depth: 4
    includes:
      before_body: header.html
      after_body:  footer.html
  pdf_document:
    extra_dependencies: ["float"]
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, include = TRUE,
                      warning = FALSE, message = FALSE,
                      root.dir = here::here(),
                      fig.width = 12, fig.height = 9)
library(here)
library(forecast.vocs)
library(dplyr)
library(loo)
library(scoringutils)
library(knitr)
library(data.table)
library(ggplot2)
library(scales)
library(dplyr)
library(here)
library(readr)
library(data.table)
library(stringr)

# load functions
source(here("R", "load-local-data.R"))
source(here("R", "plot-seq-with-cases.R"))
source(here("R", "estimate-generation-time.R"))
source(here("R", "simulations.R"))
```

```{r setup-dates}
date_data_start <- as.Date("2022-01-01")
date_gt_start <- as.Date("2022-01-12")
date_forecast_start <- as.Date("2022-02-06")
```

1. Centre for the Mathematical Modelling of Infectious Diseases, London School of Hygiene & Tropical Medicine, London WC1E 7HT, United Kingdom
2. Division for AI in Oncology, German Cancer Research Centre DKFZ, Heidelberg, Germany

## Abstract

### Background

Early estimates from South Africa indicated that the Omicron COVID-19 variant may be both more transmissible and have greater immune escape than the previously dominant Delta variant. The rapid turnover of the latest epidemic wave in South Africa as well as initial evidence from contact tracing and household infection studies has prompted speculation that the generation time of the Omicron variant may be shorter in comparable settings than the generation time of the Delta variant.

### Methods

We estimated daily growth rates for the Omicron and Delta variants in each UKHSA region from the 23rd of November to the 23rd of December 2021 using surveillance case counts by date of specimen augmented with variant type based on sequence, and genotype data with an autoregressive model that allowed for time-varying differences in the transmission advantage of the Delta variant where the evidence supported this. By assuming a gamma distributed generation distribution we then estimated the generation time distribution and transmission advantage of the Omicron variant that would be required to explain this time varying advantage. We repeated this estimation process using two different prior estimates for the generation time of the Delta variant first based on household transmission and then based on its intrinsic generation time. We also repeated this analysis using data aggregated by UKHSA region, by 15-year age group and by both UKHSA region and 15-year age group.

### Results

### Conclusions

This analysis cannot rule out the role of other factors such as differences in the populations the variants were mixing in, differences in immune escape between variants or bias due to using the test to test distribution as a proxy for the generation time distribution.

## Introduction

Early estimates from South Africa indicated that the Omicron COVID-19 variant may be both more transmissible and have greater immune escape than the previously dominant Delta variant [@pearson-omicron,@golding-omicron]. The rapid turnover of the latest epidemic wave in South Africa has prompted speculation that the generation time of the Omicron variant may be shorter in comparable settings than the generation time of the Delta variant. Recent observations of serial intervals in South Korea [@si-south-korea] supports this hypothesis though with a large degree of uncertainty and a small sample size. Other recent studies have indicated that Omicron may have a shorter incubation period than Delta, adding some potential support for this theory [@brandal-omicron-incubation]. However, the relationship between symptom presentation, immunity (and potential earlier developoment of symptoms in the presence thereof), contact behaviour (potentially modulated by self-isolation following symptoms) and generation times is far from trivial.

[Previously](https://epiforecasts.io/omicron-sgtf-forecast/summary) [@abbott_Real-time_estimation_of_2021], we estimated growth rates for the Omicron and Delta variants jointly in England by UKHSA region. We used S-gene target failure (SGTF) as a proxy of variant status (with SGTF highly correlated with Omicron) combined with reported case counts. A key finding from this report was that the growth advantage of Omicron variant versus the Delta variant appeared to vary over time. There are a range of potential explanations for this finding, such as: the impact of differential immune escape, sampling bias in the SGTF data, differences in the populations in which the variants are circulating (for example differences in age group mixtures), the impact of behavioural or policy change, or differing generation times between variants in comparable settings.

In this study, we focus on estimating the generation time distribution that would be required to explain the observed variation in growth advantage. We do so by exploiting the approximate relationship between the daily growth rate, gamma distributed generation times, and the effective reproduction number [@park2019; @park2020]. This relationship means that estimates of the growth rate for the Delta variant can be combined with prior estimates for the generation time of the Delta variant in a model in order to estimate the generation time of Omicron based on the observed variation in transmission advantage between the Omicron and Delta variant. We use growth rate estimates by UKHSA region, by 15-year age group, and by both UKHSA region and 15-year age group derived using surveillance case counts and augmented with variant type based on sequence, and genotype data along with prior estimates for the generation time from Hart et al[@hart2021].

## Methods

### Data

We use test-positive pillar 2 cases' reported variant based on sequence, and genotype data. We augment this data with total reported pillar 2 cases counts. Both sets of data were sourced from UKHSA, available daily by specimen date for England and by UKHSA region. We use data from the 23rd of November 2021 up to the 23rd of December 2021. This covers the main transition period from the Delta variant being dominant to the Omicron variant being dominant and exludes the majority of the Christmas reporting period in which test seeking behaviour became atypical. We also excluded cases with a known history of recent travel.

### Growth rate estimation

We use the same methodology to estimate variant specific growth rates as in our earlier [study estimating the time-varying transmission advantage of the Omicron variant](https://epiforecasts.io/omicron-sgtf-forecast/summary) (archived [here](https://github.com/epiforecasts/omicron-sgtf-forecast/releases/tag/reportv1.3.0)) [@abbott_Real-time_estimation_of_2021]. Briefly, we use an autoregressive (AR) approach where expected cases for each variant are modelled as a combination of expected cases from the previous day and the exponential of the log growth rate for that variant and time point. The variant specific growth rate is then itself modelled as a differenced AR(1) process using a vector autoregression structure, which assumes that the first differences of the variant growth rates are drawn from a multivariate normal distribution. This model formulation can account for variant differences other than a transmission advantage in a principled manner. Crucially, in the absence of evidence that variants do not co-vary, it reduces to a model with a fixed transmission advantage.

We fit jointly to reported cases and sequence data for each timeseries independently assuming a negative binomial and beta-binomial observation model respectively. Day of the week reporting periodicity for case counts is captured using a random effect for the day of the week. Sequence data (or its proxy) was only included in the model starting with the first date that had at least 1% of samples positive for the Omicron variant and at least 5 samples positive with the variant.

A full description of this model can be found in the documentation for the [`forecast.vocs` R package](https://epiforecasts.io/forecast.vocs/) [@forecast.vocs].

```{r load-sgtf-growth}
# get regional estimation date
regional_date <- as.Date("2022-02-12")

# load case data
daily <- load_local_data(regional_date) %>%
  filter(date <= date_forecast_start)

# load posterior estimates
sgtf <- load_results(regional_date)

sgtf_posterior <- sgtf$posterior[
  variant_relationship == "correlated"
][date <= date_forecast_start]

# load growth estimates
region_growth <- load_growth(
  regional_date, min_date = date_gt_start, max_date = date_forecast_start
)
region_growth <- region_growth[!(region %in% "England")]
```

```{r load-seq-data}
# get regional/age estimation date
age_date <- as.Date("2021-12-23")

# load case data
seq_with_cases <- read_csv(
    here("data", "public", "seq-by-region-and-age.csv")
  ) %>%
  filter(date_specimen >= date_data_start) %>%
  filter(date_specimen <= date_forecast_start)
```

```{r load-seq-results}
# load posterior estimates
seq_results <- load_results(age_date, type = "seq-by-age")

seq_posterior <- seq_results$posterior[
  variant_relationship == "correlated"
][date <= date_forecast_start]

# load growth estimates
seq_growth <- load_growth(
  age_date, min_date = date_gt_start, max_date = date_forecast_start,
  type = "seq-by-age"
)
```


### Generation time and transmission advantage estimation

We model the relationship between estimated variant growth rates, their generation times, and the between variant transmission advantage using the gamma approximation framework developed by Park et al. [@park2019; @park2020] to estimate the effective reproduction number for each variant assuming that their generation time is gamma distributed. This approximation is as follows and holds when growth is approximately exponential,

$$R = \left(1 + \bar{G}kr  \right)^{\frac{1}{k}}$$

Where $R$ is the effective reproduction number, $\bar{G}$ is the mean of the gamma distributed generation time, $r$ is the daily growth rate, and $k$ is the squared coefficient of variation which can be reparametrised as $\frac{\bar{G}^2_\mathrm{SD}}{\bar{G}^2}$ where $\bar{G}_\mathrm{SD}$ is the standard deviation of the gamma distribution.

Using this approximation we first estimate the effective reproduction number ($R^\mathrm{D}$) of the Delta variant across stratificaton ($l$) and times ($t$, in days) using estimates of the daily growth rates ($r^\mathrm{D}$), and a gamma distributed generation time. Under an assumption of homogeneous mixing we can then relate the effective reproduction number of Delta to the effective reproduction number of Omicron ($R^\mathrm{O}$) by assuming a location specific transmission advantage ($\alpha_l$) which corresponds to a combination of any intrinsic transmission advantage as well as population based advantages (such as advantage gained from immune escape). Finally, we can then reverse the gamma approximation framework to estimate the daily growth rates for Omicron ($r^\mathrm{O}$) and compare these to those estimated directly in the previous modelling step. We choose priors for our model that weakly reflect our belief that a change in generation time between variants is unlikely and if present should be relatively small in the sense that large changes of generation time are penalised.

Our model is described by the following set of equations:

\begin{align}
 r^\mathrm{D}_{lt} &\sim  \text{Normal}\left(\bar{r}^\mathrm{D}_{lt}, \tau_{lt}^\mathrm{D} \right) \\
 R^\mathrm{D}_{lt} &= \left(1 + \bar{G}^\mathrm{D}  k^\mathrm{D} r^\mathrm{D}_{lt}  \right)^{\frac{1}{k^\mathrm{D}}} \\
 R^\mathrm{O}_{lt} &= \alpha_l R^\mathrm{D}_{lt} \\
 r^\mathrm{O}_{lt} &= \frac{(R^\mathrm{O}_{lt})^{k^\mathrm{O}} - 1}{k^\mathrm{O} \bar{G}^\mathrm{O}} \\
 \sigma^\mathrm{O}_{lt} &= \sqrt{\sigma^2 + (\tau_{lt}^\mathrm{O})^2} \\
 \bar{r}^\mathrm{O}_{lt} &\sim \text{Normal}\left(r^\mathrm{O}_{lt},\sigma^\mathrm{O}_{lt} \right) \\
\end{align}

Here, superscripts D and O denote Delta and Omicron, respectively, $r$ is the growth rate, $\bar{r}$ is the mean estimated growth rate from the previous modelling step, $\tau$ its standard deviation and $R$ the effective reproduction number. We assume that generation time posteriors from the previous step are distributed according to independent normal distributions and include an additional observation error ($\sigma$) to account for the approximate nature of the relationship between variant growth rates. The relationship between generation time estimates for Omicron and Delta is then modelled as follows,

\begin{align}
  \bar{G}^\mathrm{D} &\sim \text{Normal}\left(3.2, 0.46 \right) \\
  \bar{G}_\mathrm{SD}^\mathrm{D} &\sim \text{Normal}\left(2.4, 0.33 \right) \\
  \frac{\bar{G}^\mathrm{O}}{\bar{G}^\mathrm{D}} &\sim  \text{LogNormal}\left(0, 0.2 \right) \\
  \frac{\bar{G}_\mathrm{SD}^\mathrm{O}}{\bar{G}_\mathrm{SD}^\mathrm{D}} &\sim \text{LogNormal}\left(0, 0.2 \right) \\
\end{align}

The generation priors used here are motivated by the household estimates derived in Hart et al [@hart2021] for the UK  using a normal approximation. As a sensitivity analysis, we also explore their alternative intrinsic estimates which had a mean of 4.6 days (standard deviation (SD): 0.36 days) and a standard deviation of 3.1 days (SD: 0.18) days, again using a normal approximation. We model the location specific transmission advantage ($\alpha_l$) using a random effect with all locations having a joint mean ($\alpha$) and standard deviation ($alpha_\mathrm{SD}$) as follows,

\begin{align}
  \alpha_l &\sim \text{LogNormal}\left(\alpha, \alpha_\mathrm{SD} \right) \\
  \alpha &\sim \text{Normal} \left(1, 0.5 \right) \\
  \alpha_\mathrm{SD} &\sim \text{Normal} \left(0, 0.1 \right) \\
\end{align}

Finally, we assume that the standard deviation of the normal observation model has a normal prior truncated at zero with a mean of zero and a standard deviation of 0.1.

\begin{align}
  \sigma &\sim \text{Normal} \left(0, 1 \right) \\
\end{align}

```{r load-generation-time-estimates}
posterior_summary <- fread(
  here::here("data", "retrospective", "posterior_summary.csv")
)[gt_diff == TRUE][source == "sequence"]
posterior_samples <- fread(
  here::here("data", "retrospective", "posterior_samples.csv")
)[gt_diff == TRUE][source == "sequence"]
posterior_predictions <- fread(
  here::here("data", "retrospective", "posterior_predictions.csv")
)[gt_diff == TRUE][source == "sequence"]
reproduction_no_pp <- fread(
  here::here("data", "retrospective", "reproduction_no_pp.csv")
)[gt_diff == TRUE][source == "sequence"]
posterior_locations <- fread(
  here::here("data", "retrospective", "posterior_locations.csv")
)
```

### Statistical inference

We first visualised our combined data sources (cases by specimen date and variant status by specimen date). We then fit the growth rate estimation model separately to data for each UKHSA region, 15-year age group, and both UKHSA region and 15-year age group and extracted posterior estimates of the growth rate for both Omicron and Delta, and overall. Sequence data (or its proxy) was only included in the model starting with the first date that had at least 1% of samples positive for the Omicron variant and at least 5 samples positive with the variant. This filtering was aimed at mitigating bias in the early sequence data. To motivate subsequent modelling we then visualised these estimates over time and in comparison to each other. We added linear fits to the growth rate comparison visualisations in order to visualise the relationship between variants compared to the idealised relationship in which they shared a generation time estimate. This was motivated by the simple effective reproduction number approximation $R = \text{exp} \left(\bar{G}r \right)$ . This means that the gradient of the relationship between variant growth rates can be considered as the approximate ratio between variant generation time means (and so should be 1 when they share the same generation time).

We then used the mean and standard deviation of the growth rate estimates from the 1st to the 23rd December 2021 (or from the first date with data based on our earlier filtering step), excluding early growth estimates due to the potential for bias from imported cases in the generation time and transmission advantage estimation modelling step. We repeated this estimation process for growth rate estimates by UKHSA region, by 15-year age group, and for both jointly. Each of these models assumed a shared generation time distribution for Omicron across all growth rate estimates but the tranmission advantage was able to vary though this was constrained using a hyperparameter. We used both the household and intrinsic Delta generation time estimates from Hart et al. [@hart2021] and presented the resulting posterior estimates from both. To check the performance of our modelling approach, we plot the realised generation time distributions our posterior estimates implied, plot the scatter plots of the mean and standard deviation of the generation time, plot the stratification specific transmission advantange, and plot the posterior predictions for Omicron daily growth rates over time and compared them to the estimates from the growth rate estimation step.

### Implementation

The growth rate estimation model was implemented using the [`forecast.vocs` R package](https://epiforecasts.io/forecast.vocs/) [@R; @forecast.vocs] and fit using `stan` [@stan] and `cmdstanr` [@cmdstanr]. The model was fit using 2 chains with each chain having 1000 warmup steps and 2000 sampling steps. The generation time estimation model was also implemented using `stan` [@stan] and `cmdstanr` [@cmdstanr] with the default of 4 chains each with 1000 warmup and 1000 sampling steps. For all models convergence was assessed using the Rhat diagnostic [@stan] and fitting was initialised using samples drawn from near the mean of each prior distributions.

## Results

### Scenario simulations

```{r syn-scenario-plot, fig.height = 9, fig.width = 9, fig.cap = "Growth rates scenarios based on an observed growth rate for one variant (black line) and the resulting growth rate for another variant varying the assumed transmission advantage (coloured lines), generation time mean (x facet - 75%, 100% and 125% of the baseline assumed value of 3.2 days), and generation time standard deviation (y axis, 75%, 100% and 125% of the baseline assumed value of 2.1 days). The central facet represents a scenario where variants share a generation time."}
syn_sims <- fread(here("data", "simulations", "uk-scenario-simulations.csv"))
plot_syn_scenarios(syn_sims[gt_type %in% "Household"])
```

### Data description

```{r data, fig.cap = "Daily pillar 2 cases by specimen date in England and by UKHSA region with variant type (Omicron, non-Omicron, and unknown), starting from the 23rd of November 2021 and including data up to the 23rd of December 2021."}
seq_with_cases %>%
  filter(age_group %in% "Overall") %>%
  group_by(region, date_specimen) %>%
  summarise(across(where(is.numeric), sum), .groups = "drop") %>%
  plot_seq_with_cases() +
    facet_wrap(~ region, scales = "free")
```

```{r data-by-age, fig.cap = "Daily pillar 2 cases by specimen date in England by 15-year age group with variant type (Omicron, non-Omicron, and unknown), starting from the 23rd of November 2021 and including data up to the 23rd of December 2021."}
seq_with_cases %>%
  filter(!(age_group %in% "Overall")) %>%
  filter(region %in% "England") %>%
  plot_seq_with_cases() +
    facet_wrap(~ age_group, scales = "free")
```

```{r data-by-age-regions, fig.cap = "Daily pillar 2 cases by specimen date in England by UKHSA region 15-year age group with variant type (Omicron, non-Omicron, and unknown), starting from the 23rd of November 2021 and including data up to the 23rd of December 2021."}
seq_with_cases %>%
  filter(!(region %in% "England")) %>%
  plot_seq_with_cases() +
  facet_grid(age_group ~ region, scales = "free_y")
```

### Growth rate of reported cases overall, with the Omicron variant, and not with the Omicron variant

#### Over time

```{r growth, fig.cap = "Estimates of the time-varying daily growth rate for England and by UKHSA region (using the log scale approximation) for reported cases overall, cases with the Omicron variant, and non-Omicron cases. Growth rates are assumed to be piecewise constant with potential changes every 3 days. Growth rates are assumed to be constant beyond the forecast horizon (vertical dashed line). 90% and 60% credible intervals are shown."}
plot_growth(
  seq_posterior[age_group %in% "Overall"]
) +
  facet_wrap(~ region)
```

```{r age-growth, fig.cap = "Estimates of the time-varying daily growth rate for England by 15-year age group (using the log scale approximation) for reported cases overall, cases with the Omicron variant, and non-Omicron cases. Growth rates are assumed to be piecewise constant with potential changes every 3 days. Growth rates are assumed to be constant beyond the forecast horizon (vertical dashed line). 90% and 60% credible intervals are shown."}
plot_growth(
  seq_posterior[region %in% "England"][!(age_group %in% "Overall")]) +
  facet_wrap(~ age_group)
```

```{r age-region-growth, fig.cap = "Estimates of the time-varying daily growth rate for England by UKHSA region and 15-year age group (using the log scale approximation) for reported cases overall, cases with the Omicron variant, and non-Omicron cases. Growth rates are assumed to be piecewise constant with potential changes every 3 days. Growth rates are assumed to be constant beyond the forecast horizon (vertical dashed line). 90% and 60% credible intervals are shown."}
plot_growth(
  seq_posterior[!(region %in% "England")][!(age_group %in% "Overall")]
) +
  facet_grid(age_group ~ region)
```

#### Omicron versus Delta

```{r plot-o-vs-d-growth, fig.cap = "Estimates of the time-varying daily growth rate in England by UKHSA region for cases with the Omicron variant, and non-Omicron cases. Point size has been used to represent the uncertainty of each estimate with smaller points being more uncertain. The dashed grey line indicates the expected relationship between variants if they differ only by a fixed transmission advantage. The solid grey line indicates the observed overall linear relationship between variants with the slope of this line approximating the ratio between generation time means. The region specific lines show how this relationship differs between regions."}
piv_growth <- seq_growth %>%
  filter(age_group %in% "Overall", !(region %in% "England")) %>%
  pivot_wider(id_cols = c("region", "date"),
              names_from = "type", values_from = c("mean", "sd")) %>%
  filter(!is.na(mean_Omicron))
min_nomicron <- mean(piv_growth[["mean_Omicron"]]) -
  mean(piv_growth[["mean_non-Omicron"]])

ggplot(piv_growth) +
  aes(y = mean_Omicron, x = .data[["mean_non-Omicron"]], size = 1 / sd_Omicron,
      col = region
  ) +
  scale_color_brewer(palette = "Paired") +
  geom_abline(slope = 1, intercept = min_nomicron, col = "#777575",
              linetype = 2, size = 1.5) +
  geom_smooth(method = "lm", formula = y ~ x,
              se = FALSE, alpha = 0.2) +
  geom_smooth(col = "#777575", method = "lm", formula = y ~ x,
              se = FALSE, size = 1.5, alpha = 0.3, linetype = 1) +
  geom_point(alpha = 0.3) +
  scale_size(guide = "none") +
  theme_bw() +
  theme(legend.position = "bottom") +
  labs(x = "Delta daily growth rate", y = "Omicron daily growth rate",
       col = "UKHSA region")
```

```{r plot-o-vs-d-growth-age, fig.cap = "Estimates of the time-varying daily growth rate in England by 15-year age group for cases with the Omicron variant, and non-Omicron cases. Point size has been used to represent the uncertainty of each estimate with smaller points being more uncertain. The dashed grey line indicates the expected relationship between variants if they differ only by a fixed transmission advantage. The solid grey line indicates the observed overall linear relationship between variants with the slope of this line approximating the ratio between generation time means."}
piv_growth <- seq_growth %>%
  filter(!(age_group %in% "Overall"), region %in% "England") %>%
  pivot_wider(id_cols = c("age_group", "date"),
              names_from = "type", values_from = c("mean", "sd"))  %>%
  filter(!is.na(mean_Omicron))
min_nomicron <- mean(piv_growth[["mean_Omicron"]]) -
  mean(piv_growth[["mean_non-Omicron"]])

ggplot(piv_growth) +
  aes(y = mean_Omicron, x = .data[["mean_non-Omicron"]], size = 1 / sd_Omicron,
      col = age_group
  ) +
  scale_color_brewer(palette = "Paired") +
  geom_abline(slope = 1, intercept = min_nomicron, col = "#777575",
              linetype = 2, size = 1.5) +
  geom_smooth(method = "lm", formula = y ~ x,
              se = FALSE, alpha = 0.2) +
  geom_smooth(col = "#777575", method = "lm", formula = y ~ x,
              se = FALSE, size = 1.5, alpha = 0.3, linetype = 1) +
  geom_point(alpha = 0.3) +
  scale_size(guide = "none") +
  theme_bw() +
  theme(legend.position = "bottom") +
  labs(x = "Delta daily growth rate", y = "Omicron daily growth rate",
       col = "Age group")
```

```{r plot-o-vs-d-growt-age-region, fig.cap = "Estimates of the time-varying daily growth rate in England by UKHSA region and 15-year age group for cases with the Omicron variant, and non-Omicron cases. Point size has been used to represent the uncertainty of each estimate with smaller points being more uncertain. The dashed grey line indicates the expected relationship between variants if they differ only by a fixed transmission advantage. The solid grey line indicates the observed overall linear relationship between variants with the slope of this line approximating the ratio between generation time means."}
piv_growth <- seq_growth %>%
  filter(!(age_group %in% "Overall"), !region %in% "England") %>%
  pivot_wider(id_cols = c("region", "age_group", "date"),
              names_from = "type", values_from = c("mean", "sd"))  %>%
  filter(!is.na(mean_Omicron))
min_nomicron <- mean(piv_growth[["mean_Omicron"]]) -
  mean(piv_growth[["mean_non-Omicron"]])

ggplot(piv_growth) +
  aes(y = mean_Omicron, x = .data[["mean_non-Omicron"]], size = 1 / sd_Omicron,
      col = region, shape = age_group
  ) +
  scale_color_brewer(palette = "Dark2") +
  geom_abline(slope = 1, intercept = min_nomicron, col = "#777575",
              linetype = 2, size = 1.5) +
  geom_smooth(col = "#777575", method = "lm", formula = y ~ x,
              se = FALSE, size = 1.5, alpha = 0.3, linetype = 1,
              aes(shape = NA)) +
  geom_point(alpha = 0.3) +
  scale_size(guide = "none") +
  theme_bw() +
  theme(legend.position = "bottom") +
  labs(x = "Delta daily growth rate", y = "Omicron daily growth rate",
       col = "UKHSA region", shape = "Age group")
```

### Generation time and transmission advantage estimates

```{r, fig.height = 9, fig.width = 12, fig.cap = "Posterior samples of the generation time distribution for both Omicron and Delta variants, each generation time prior assumption and data stratification. 100 sampled generation time distributions are shown."}
gt_samples <- posterior_samples %>%
  filter(.draw < (250 / max(.chain))) %>%
  select(stratification, gt_type, gt_mean, gt_sd, voc_gt_mean, voc_gt_sd) %>%
  group_by(stratification, gt_type) %>%
  mutate(sample = 1:n()) %>%
  pivot_longer(gt_mean:voc_gt_sd, names_to = c("variant", ".value"),
               names_pattern = "(.+)_(.+)") %>%
  mutate(variant = ifelse(variant %in% "voc_gt", "Omicron", "Delta")) %>%
  mutate(stratification = str_to_sentence(stratification),
         gt_type = str_to_sentence(gt_type))

gt_samples %>%
  mutate(
    pmf = purrr::map2(
      mean, sd,
      ~ tibble(x = 0:30,
      prob = c(0, pgamma(1:30, (.x / .y)^2, .x / .y^2) -
        pgamma(0:29, (.x / .y)^2, .x / .y^2)))
    )
  ) %>%
  unnest(pmf) %>%
  filter(sample <= 100) %>%
  filter(x <= 15) %>%
  ggplot() +
  aes(x = x, y = prob, col = variant, group = interaction(sample, variant)) +
  geom_line(alpha = 0.3) +
  theme_bw() +
  theme(legend.position = "bottom") +
  scale_color_brewer(palette = "Dark2") +
  labs(y = "Probability density", x = "Day") +
  facet_grid(stratification~gt_type)
```

```{r, fig.height = 9, fig.width = 9, fig.cap = "Posterior samples of the generation time distribution mean and standard deviation for both Omicron and Delta variants, each generation time prior assumption and data stratification. 250 samples are shown." }
gt_samples %>%
  ggplot() +
  aes(x = sd, y = mean, col = variant) +
  geom_point(alpha = 0.2) +
  theme_bw() +
  theme(legend.position = "bottom") +
  scale_color_brewer(palette = "Dark2") +
  labs(y = "Generation time mean", x = "Generation time standard deviation") +
  facet_grid(stratification~gt_type)
```

```{r}
clean_summary <- posterior_summary %>%
  select(-rhat, -ess_bulk, -ess_tail) %>%
  filter(
    variable %in% c("gt_mean", "gt_sd", "ta", "ta_sd", "voc_gt_mean",
                    "voc_gt_sd")
  ) %>%
  mutate(across(where(is.numeric),
                ~ ifelse(variable %in% c("ta"),
                  exp(.), .))
  ) %>
  mutate(across(where(is.numeric), round, digits = 1)) %>%
  mutate(variant = case_when(variable %in% c("gt_mean", "gt_sd") ~ "Omicron BA.1",
                             TRUE ~ "Omicron BA.2")) %>%
  mutate(stratification = str_to_sentence(stratification),
         gt_type = str_to_sentence(gt_type)) %>%
  select(data_stratification = stratification,
         `Generation time prior` = gt_type, variant,
         everything()) %>%
  mutate(variable = case_when(
    variable %in% "ta" ~ "Transmission advantage",
    variable %in% "ta_sd" ~ "Transmission advantage standard deviation (log)",
    variable %in% c("gt_mean", "voc_gt_mean") ~ "Generation time mean",
    TRUE ~ "Generation time standard deviation"
    )
  ) %>%
  janitor::clean_names(case = "sentence") %>%
  mutate(across(where(is.character), as.factor)) %>%
  rename(SD = Sd, MAD = Mad)
```

```{r, fig.width = 12, fig.height = 6, fig.cap = "Median and 90% credible intervals of the posterior estimates for the generation time distribution mean and standard deviation, the transmission advantage, and the standard deviation of the transmission advantage for both Omicron and Delta variants, each generation time prior assumption and data stratification." }
clean_summary %>%
  ggplot() +
  aes(x = `Data stratification`, y = Median, ymin = Q5, ymax = Q95,
      col = Variant) +
  geom_point(size = 2, position = position_dodge(width = 0.4)) +
  geom_linerange(size = 2, position = position_dodge(width = 0.4),
                 alpha = 0.8) +
  coord_flip() +
  facet_grid(`Generation time prior` ~ Variable, scales = "free_x") +
  theme_bw() +
  theme(legend.position = "bottom") +
  scale_color_brewer(palette = "Paired") +
  labs(y = "")
```

```{r, eval = FALSE}
posterior_summary %>%
  filter(
    variable %in% c("ta", "local_ta")
  ) %>%
  mutate(across(where(is.numeric),
                ~ ifelse(variable %in% c("ta", "local_ta"),
                  exp(.), .))
  ) %>%
  ggplot() +
  aes(x = `Data stratification`, y = Median, ymin = Q5, ymax = Q95,
      col = Variant) +
  geom_point(size = 2, position = position_dodge(width = 0.4)) +
  geom_linerange(size = 2, position = position_dodge(width = 0.4),
                 alpha = 0.8) +
  coord_flip() +
  facet_grid(`Generation time prior` ~ Variable, scales = "free_x") +
  theme_bw() +
  theme(legend.position = "bottom") +
  scale_color_brewer(palette = "Paired") +
  labs(y = "")
```

```{r}
clean_summary %>%
  mutate(across(where(is.numeric), round, digits = 1)) %>%
  kable(caption = "Summarised posterior estimates for generation time and transmission advantage by data stratification, and choice of generation time prior")
```

### Posterior predictions

#### Growth rates

```{r pp-region, fig.cap = "Estimates of the time-varying daily growth rate for each UKHSA region for cases with the Omicron variant compared to posterior predictions using Delta growth rates and differing generation times. Delta growth rates are assumed to be piecewise constant with potential changes every 3 days. 90% and 60% credible intervals are shown."}
posterior_predictions %>%
  filter(stratification == "region", gt_type == "household") %>%
  gt_plot_pp() +
  facet_wrap(~region)
```

```{r pp-age, fig.cap = "Estimates of the time-varying daily growth rate for each 15 year age group for cases with the Omicron variant compared to posterior predictions using Delta growth rates and differing generation times. Delta growth rates are assumed to be piecewise constant with potential changes every 3 days. 90% and 60% credible intervals are shown.", fig.width = 9, fig.height = 9}
posterior_predictions %>%
  filter(stratification == "age", gt_type == "household") %>%
  gt_plot_pp() +
  facet_wrap(~age_group)
```

```{r pp-age-region, fig.cap = "Estimates of the time-varying daily growth rate for each UKHSA region and 15-year age group for cases with the Omicron variant compared to posterior predictions using Delta growth rates and differing generation times. Delta growth rates are assumed to be piecewise constant with potential changes every 3 days. 90% and 60% credible intervals are shown."}
posterior_predictions %>%
  filter(stratification == "age and region", gt_type == "household") %>%
  gt_plot_pp() +
  facet_grid(age_group~region)
```

#### Reproduction number estimates

```{r rt-pp-region, fig.cap = "Estimates of the time-varying effective reproduction number for each UKHSA region for cases with the Omicron variant compared to posterior predictions using Delta growth rates and differing generation times. Delta growth rates are assumed to be piecewise constant with potential changes every 3 days. 90% and 60% credible intervals are shown.", fig.width = 9, fig.height = 6}
reproduction_no_pp %>%
  filter(stratification == "region") %>%
  gt_plot_pp(
    fill = "region", fill_lab = "UKSHA region",
    y = "Effective reproduction number", yint = 1
  ) +
  facet_wrap(~gt_type)
```

```{r rt-pp-age, fig.cap = "Estimates of the time-varying effective reproduction number for each 15 year age group for cases with the Omicron variant compared to posterior predictions using Delta growth rates and differing generation times. Delta growth rates are assumed to be piecewise constant with potential changes every 3 days. 90% and 60% credible intervals are shown.", fig.width = 9, fig.height = 6}
reproduction_no_pp %>%
  filter(stratification == "age") %>%
  gt_plot_pp(
    fill = "age_group", fill_lab = "Age group",
    y = "Effective reproduction number", yint = 1
  ) +
  facet_wrap(~gt_type)
```

```{r rt-pp-age-region, fig.cap = "Estimates of the time-varying effective reproduction number for each UKHSA region and 15-year age group for cases with the Omicron variant compared to posterior predictions using Delta growth rates and differing generation times. Delta growth rates are assumed to be piecewise constant with potential changes every 3 days. 90% and 60% credible intervals are shown."}
reproduction_no_pp %>%
  filter(stratification == "age and region") %>%
  gt_plot_pp(
    fill = "age_group", fill_lab = "Age group",
    y = "Effective reproduction number", yint = 1
  ) +
  facet_grid(gt_type~region)
```

## Discussion

Our analysis may be biased by a range of factors and so these findings should be considered in the light of other results.

Our analysis had several key limitations. Firstly, the surveillance data sources we used to estimate daily growth rates had the potential to bias our analysis both because sequence data may be unrepresentative due to sampling strategy and because we used public case count data which do not include reinfections. The exclusion of known reinfections could also lead to bias in our estimates if the proportion of infections that are reinfections varies over time and between variants. Changes in testing practice may also impact our results, especially if this occurs differentially in populations where different proportions of each variant are circulating. In our first study[@abbott_Real-time_estimation_of_2021], we used S-gene status as a proxy for variant status and found similar results to those presented here by UKSHA region. We also explored the potential for sampling bias in this study for cases with known S-gene status and found little evidence that this was present.

Our generation time and transmission advantage estimates for Omicron also have limitations. A major limitation is that our estimated generation time distributions are in fact the distributions from a positive test to resulting positive tests rather than the true generation time distribution which would be based on infection times. This means that our estimates may be biased if those secondary infections test positive prior to the original (primary) infected case, or may be biased by the epidemic phase [@volz2021]. In addition to this, our model only estimates a location-specific overall transmission advantage. We do not attempt to distinguish between "intrinsic" transmissibility and immune escape, a combination of which is likely to be underlying any sustained differences in growth rate and to potentially vary over time due to increases in population immunity. It is also possible that the generation time differs within UKSHA regions and by age group due to differences in population structure, prior immunity and other factors. However, our posterior predictive model checks imply a good match between observed and estimated growth rates for Omicron for the majority of data points indicating that this effect may be minimal. Finally, our estimates are based on aggregate populations with the smallest scale we consided being 15-year age groups by UKHSA region. There may be additional variation within these groups that these estimates do not account for and this may bias our results.

There are several alternative explanations for our findings other than a change in the generation time distribution. One potential explanation is that that Omicron and Delta may have had differing age profiles in the time period studied, these age distributions may have changed over time, and age groups may have differing transmission potentials. This could potentially lead to the observed variation in transmission advantage between Omicron and Delta without requiring differing generation times. Similarly, if age groups change behaviour differently over time or respond differently to policy changes this could lead to similar changes in transmission advantage as observed.

Our results may also be the result of differential immune escape between variants. The reduction in the observed transmission advantage of Omicron over time could also be the result of Delta having greater immune escape from previous Omicron infection than Omicron. However, this seems unlikely given epidemiological evidence [@ferguson-omicron-transmission] and the short time lines involved. As observed in Volz et al. social distancing typically reduces the number of people contacted per day, but increases the duration and frequency of household contacts [@volz2021]. This can result in the saturation of available contacts and hence reduce estimates of the transmission advantage for a variant. If social contacts changed markedly during our study period this could also be an explanation for the reduction over time in the observed transmission advantage of Omicron. Finally, the variants may have been circulating in populations that differed in more than just the distribution of ages. This would likely have a similar impact in terms of biasing our results. 

Despite the range of potential sources of bias our results do correspond with observations of reduced serial intervals in South Korea [@si-south-korea] and a potentially reduced incubation period [@brandal-omicron-incubation] although both these studies reported a large degree of uncertainty and the relationship between symptom presentation, immunity, contact behaviour and generation times is far from trivial.

Our results may also be the result of differential immune escape between variants. The reduction in the observed transmission advantage of Omicron over time could result from Delta having greater immune escape from previous Omicron infection than Omicron. However, this seems unlikely given epidemiological evidence [@ferguson-omicron-transmission] and the short time lines involved. As observed in Volz et al. social distancing typically reduces the number of people contacted per day, but increases the duration and frequency of household contacts [@volz2021]. This can result in the saturation of available contacts and hence reduce estimates of the transmission advantage for a variant. If social contacts changed markedly during our study period this could be an explanation for the reduction over time in the observed transmission advantage of Omicron. Finally, the variants may have been circulating in populations that differed in more than just the distribution of ages. This would likely have a similar impact in terms of biasing our results. Despite the range of potential sources of bias our results do correspond with observations of reduced serial intervals in South Korea [@si-south-korea] and a potentially reduced incubation period [@brandal-omicron-incubation], although both these studies reported a large degree of uncertainty and the relationship between symptom presentation, immunity, contact behaviour and generation times is far from trivial.

This analysis cannot rule out the role of other factors such as differences in the populations the variants were mixing in, differences in immune escape between variants, or a bias due to using the test-to-test distribution as a proxy for the generation time. More work is needed to explore these factors both using similar methods to those described in this study, potentially extending to a latent infection based model, and also using alternative study designs such as household transmission studies and contact tracing studies.

## References
